src/Symfony/Component/PropertyAccess/PropertyAccessor.php

```diff
@@ -304,7 +304,7 @@ private function writeProperty(&$object, $property, $singular, $value)
                 // see https://github.com/symfony/symfony/issues/4670
                 $itemsToAdd = is_object($value) ? iterator_to_array($value) : $value;
                 $itemToRemove = array();
-                $propertyValue = $this->readProperty($object, $property);
+                $propertyValue = &$this->readProperty($object, $property);
```

Comment:
I suggest unsetting the ``$propertyValue`` just after it is not needed, to avoid future mistake by reassigning to it